src/fontra/views/editor/edit-tools-pointer.js

```diff
@@ -346,6 +405,179 @@ export class PointerTool extends BaseTool {
+    // Because we say for example bottom-left and not left-bottom. Y-X order.
+    const resizeHandlePositionX = clickedResizeHandle.split("-")[1];
+    const resizeHandlePositionY = clickedResizeHandle.split("-")[0];
+
+    const origin = { x: resizeHandlePositionX, y: resizeHandlePositionY };
+    // origin must be the opposite side of where we have our mouse
+    if (resizeHandlePositionX === "left") {
+      origin.x = "right";
+    }
+    if (resizeHandlePositionX === "right") {
+      origin.x = "left";
+    }
+    if (resizeHandlePositionY === "top") {
+      origin.y = "bottom";
+    }
+    if (resizeHandlePositionY === "bottom") {
+      origin.y = "top";
+    }
+    // no else because could be middle or center
+
+    // must be set to the opposite side of the mouse if left or bottom
+    const fixDragLeftValue = clickedResizeHandle.includes("left") ? -1 : 1;
+    const fixDragBottomValue = clickedResizeHandle.includes("bottom") ? -1 : 1;
+
+    const staticGlyphControllers = await sceneController.getStaticGlyphControllers();
+
+    await sceneController.editGlyph(async (sendIncrementalChange, glyph) => {
+      const initialPoint = sceneController.selectedGlyphPoint(initialEvent);
+
+      const layerInfo = Object.entries(
+        sceneController.getEditingLayerFromGlyphLayers(glyph.layers)
+      ).map(([layerName, layerGlyph]) => {
+        const behaviorFactory = new EditBehaviorFactory(
+          layerGlyph,
+          sceneController.selection,
+          sceneController.experimentalFeatures.scalingEditBehavior
+        );
+        const layerBounds =
+          staticGlyphControllers[layerName].getSelectionBounds(selection);
+
+        return {
+          layerName,
+          changePath: ["layers", layerName, "glyph"],
+          layerGlyphController: staticGlyphControllers[layerName],
+          editBehavior: behaviorFactory.getBehavior("default", true),
+          regularPinPoint: getPinPoint(layerBounds, origin.x, origin.y),
+          altPinPoint: getPinPoint(layerBounds, undefined, undefined),
+          selectionWidth: layerBounds.xMax - layerBounds.xMin,
+          selectionHeight: layerBounds.yMax - layerBounds.yMin,
+        };
+      });
+
+      let editChange;
+      for await (const event of eventStream) {
+        const currentPoint = sceneController.selectedGlyphPoint(event);
+        const delta = {
+          x: (currentPoint.x - initialPoint.x) * fixDragLeftValue,
+          y: (currentPoint.y - initialPoint.y) * fixDragBottomValue,
+        };
+
+        const deepEditChanges = [];
+        for (const layer of layerInfo) {
+          const layerGlyph = layer.layerGlyphController.instance;
+
+          // NOTE: calculate the scale based on selection width per layer.
+          let scaleX = (layer.selectionWidth + delta.x) / layer.selectionWidth;
+          let scaleY = (layer.selectionHeight + delta.y) / layer.selectionHeight;
+
+          if (clickedResizeHandle.includes("middle")) {
```

Comment:
This if block plus the two following if blocks should be if / else if / else if, no?

Reply:
I would say no, because we have calculated scaleX and scaleY already before. An `else` would fallback to what we have already in line 479 and 480. But let me think about it a bit more.

Reply:
Aren't the three blocks mutually exclusive? If one "if" triggers, none of the others will. Or do I miss something?

Reply:
If "middle", there will not be "center", and the last shiftKey "if" only needs to be done if there is no "middle" and no "center".

Reply:
Done + further code improvements